### DISCUSSION TITLE
Reactive UI vs MVVM

### DISCUSSION THREAD

**Discussion by Yveske1966** [AvaloniaUI/Avalonia#13837] [General]
I'm new to C# (coming from CBuilder) and made my first steps in Maui with MVVM.
Avalonia documentation uses Reactive UI, which I find very verbose, but I'm willing to learn.
I know I can stay with MVVM but wanted to know if there is any advantage to using Reactive UI.
Is there a "best used in case of" I need to know before I regret down the road ?

**Comment by stevemonaco** [comment]
ReactiveUI has out-of-the-box support for reactive programming with helpers beyond what System.Reactive and DynamicData have. It also has built-in support for routing and a few other things that Mvvm Toolkit doesn't have.

You can add reactivity to Mvvm Toolkit through adding System.Reactive, ReactiveProperty, and DynamicData libraries but you'll still lack some of the helpers. You can even use Mvvm Toolkit and ReactiveUI side-by-side in certain scenarios to fill in gaps as long as you're careful about it.

**Comment by Yveske1966** [comment]
Thanks !
I think I will stick with MVVM for now as my first app is just administrative.
Will look into Reactive when I get around doing some stuff with Raspberry Pi.

**Reply by timunie** [reply]
Fyi: when Avalonia started, the MVVM Toolkit wasn't there and ReactiveUI was the best in our opinion. Nowadays both works well and also side by side if needed. Unless you need some feature, I'd stick with the MVVM Toolkit 😊. 

So the docs showing RxUI is mainly for historical reasons.

Welcome and happy coding
